### PR TITLE
Plugin facelift towards the 1.4 release

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This change updates the plugin to the newer parent POM towards the 1.4 release. There was no updates for a long time, hence there are pretty much changes to be done.

- [x] - Update the minimal Jenkins core version to 1.609.3 - the last version with Java 6 support
- [x] - Update to Parent POM 2.x
- [x] - Cleanup FindBugs issues. There are valid ones in the file listing logic
- [x] - Escape all Jelly files by default
- [x] - Fix binary incompatibility introduces in non-released changes
- [x] - Cleanup some ancient plugin definitions (probably others can be cleaned up as well)
- [x] - Get rid of ancient dependencies explicitly defined in POM
- [x] - Make `mvn site` really working with Maven 3

@reviewbybees, cc @svanoort who was looking at this code before

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -73,45 +63,36 @@
...
 
     <properties>
+        <!--For parent POM-->
+        <jenkins.version>1.609.3</jenkins.version>
+        <java.level>6</java.level>
+        
         <!-- define all plugin versions -->
-        <maven.version>3.0.1</maven.version>
         <maven-antrun-plugin.version>1.6</maven-antrun-plugin.version>
         <maven-assembly-plugin.version>2.2</maven-assembly-plugin.version>
```
Comment (oleg-nenashev) [inline pom.xml]: I'd guess much other stuff can be removed as well

File: pom.xml [hunk 2]
```diff
@@ -37,11 +37,6 @@
 
     <dependencies>
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
-            <version>4.11</version>
-        </dependency>
-        <dependency>
             <groupId>org.codehaus.plexus</groupId>
```
Comment (oleg-nenashev) [inline pom.xml]: This archiver is also ancient, but I do not want to touch it now

File: src/main/java/org/jenkinsci/plugins/periodicbackup/ConfigOnly.java [hunk 3]
```diff
@@ -101,11 +108,16 @@ private void addJobFiles(File rootDir, List<File> filesToBackup) {
...
 
-    private void addUserFiles(File rootDir, List<File> filesToBackup) {
+    private void addUserFiles(File rootDir, List<File> filesToBackup) throws PeriodicBackupException {
         File usersDir = new File(rootDir, "users");
         if (usersDir.exists() && usersDir.isDirectory()) {
             // Each user directory should have a config.xml file
             File[] dirsInUsers = usersDir.listFiles((FileFilter) FileFilterUtils.directoryFileFilter());
+            if (dirsInUsers == null) {
+                // First one may happen only due to the race condition
+                throw new PeriodicBackupException("Job directory path does not denote or there is an I/O error");
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/periodicbackup/ConfigOnly.java]: 🐛  "Does not denote" --> Invalid Job directory path or I/O error

File: src/main/java/org/jenkinsci/plugins/periodicbackup/Util.java [hunk 4]
```diff
@@ -156,4 +160,44 @@ public static String getExtension(File f) {
...
+    public static File[] listFiles(@Nonnull File directory, @CheckForNull FileFilter fileFilter) throws PeriodicBackupException {
+        if (!directory.isDirectory()) {
+            throw new PeriodicBackupException(formatMessage(directory, "File is not a directory"));
+        }
+        
+        final File[] files;
+        try {
+            files = fileFilter == null ? directory.listFiles() : directory.listFiles(fileFilter);
+        } catch(SecurityException ex) {
+            throw new PeriodicBackupException(formatMessage(directory, "Securiy exception while listing files"), ex);
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/periodicbackup/Util.java]: 🐛  "Securiy"

**Comment by oleg-nenashev** [comment]
@svanoort done

File: src/main/java/org/jenkinsci/plugins/periodicbackup/ConfigOnly.java [hunk 5]
```diff
@@ -112,12 +112,7 @@ private void addUserFiles(File rootDir, List<File> filesToBackup) throws Periodi
         File usersDir = new File(rootDir, "users");
         if (usersDir.exists() && usersDir.isDirectory()) {
             // Each user directory should have a config.xml file
-            File[] dirsInUsers = usersDir.listFiles((FileFilter) FileFilterUtils.directoryFileFilter());
-            if (dirsInUsers == null) {
-                // First one may happen only due to the race condition
-                throw new PeriodicBackupException("Job directory path does not denote or there is an I/O error");
-            }
-            
+            File[] dirsInUsers =  Util.listFiles(usersDir, FileFilterUtils.directoryFileFilter());
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/periodicbackup/ConfigOnly.java]: @oleg-nenashev  ??

File: src/main/java/org/jenkinsci/plugins/periodicbackup/Util.java [hunk 6]
```diff
@@ -183,7 +183,7 @@ public static boolean isWritableDirectory(File directory) {
         try {
             files = fileFilter == null ? directory.listFiles() : directory.listFiles(fileFilter);
         } catch(SecurityException ex) {
-            throw new PeriodicBackupException(formatMessage(directory, "Securiy exception while listing files"), ex);
+            throw new PeriodicBackupException(formatMessage(directory, "Security exception while listing files"), ex);
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/periodicbackup/Util.java]: 👍

File: src/main/java/org/jenkinsci/plugins/periodicbackup/ConfigOnly.java (hunk 5 above)
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/periodicbackup/ConfigOnly.java]: Hm? Just replaced it by the `Util` method which I've created. A leftover from the original fix attempts

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Comment by oleg-nenashev** [comment]
Going forward with the release

**Pull request merged**